File: Algorithms/src/3643. Flip Square Submatrix Vertically/Code/solution.js
var reverseSubmatrix = function (grid, x, y, k) {
  // Loop half rows
  for (let i = 0; i < Math.floor(k / 2); i++) {
    let top = x + i;
    let bottom = x + k - 1 - i;

    // Swap columns
    for (let j = 0; j < k; j++) {
      let temp = grid[top][y + j];
      grid[top][y + j] = grid[bottom][y + j];
      grid[bottom][y + j] = temp;
    }
  }
  return grid;
};

File: Algorithms/src/3643. Flip Square Submatrix Vertically/Code/solution.go
func reverseSubmatrix(grid [][]int, x int, y int, k int) [][]int {
    // Loop half rows
    for i := 0; i < k/2; i++ {
        top := x + i
        bottom := x + k - 1 - i

        // Swap columns
        for j := 0; j < k; j++ {
            grid[top][y+j], grid[bottom][y+j] = grid[bottom][y+j], grid[top][y+j]
        }
    }
    return grid
}